app/models/manageiq/providers/vmware/infra_manager/inventory/collector.rb

```diff
@@ -311,13 +311,32 @@ def parse_content_libraries(api_client, parser)
 
     library_api.list.value.each do |lib_id|
       library_item_api.list(lib_id).value.each do |item_id|
-        parser.parse_content_library_item(library_item_api.get(item_id).value)
+        library_item = library_item_api.get(item_id).value
+        next unless library_item.type == "ovf"
+
+        content = query_ovf_template(api_client, item_id)
+        parser.parse_content_library_item(library_item, content)
       end
     end
   rescue VSphereAutomation::ApiError
     nil
   end
 
+  def query_ovf_template(api_client, ovf_library_item_id)
+    require 'vsphere-automation-vcenter'
+
+    resource_pool = ems.resource_pools.where(:vapp => false).first
```

Comment:
:-1: for this kind of thing in the collector

1. There is no guarantee that the resource pool has been created yet (think initial refresh, there will be 0)
2. What is special about the first resource pool?  Multiple clusters/datacenters etc...

Why do you need a specific resource pool?

Reply:
A resource pool is the required field for the query. 
Maybe this query should be performed when creating the service catalog item?

Reply:
Yeah if the target is required for this we can't possibly do this at refresh time, only other time would be during deployment.